File: handle_directory/handle.js
var fs = require("fs");
var path = require("path");
var rootPath = path.resolve();
var filePath = path.resolve() + '/assets';
var fileArr = [];
var fileObj = {};

let rootArr = [];

fs.readdir(filePath, function (err, files) {
    if (err) {
        console.log(err);
        return;
    }
    files.forEach(function (filename) {
        fs.stat(path.join(filePath, filename), function (err, stats) {
            if (err) throw err;
            //文件
            if (stats.isFile()) {
                console.log(filename);
            } else if (stats.isDirectory()) {
                console.log(filename);
                readFile(path.join(filePath, filename), filename, rootArr, already);
            }
        });
    });
});

//获取文件数组
function readFile(readurl, name, testArr, next) {
    // console.log(name);
    let testObj = {
        title: name,
        urlList: [],
        titleList: []
    };
    testArr.push(testObj);
    fs.readdir(readurl, function (err, files) {
        if (err) { console.log(err); return; }

        files.forEach(function (filename) {
            let realFileName = path.join(readurl, filename);
            fs.stat(realFileName, function (err, stats) {
                if (err) throw err;
                //是文件
                if (stats.isFile()) {
                    testObj.urlList.push(realFileName);
                    writeFile(rootArr);
                } else if (stats.isDirectory()) {
                    readFile(realFileName, filename, testObj.titleList, already);
                }
            });
        });
    });
}

// already

function already(){
    console.log(fileObj);
}

// 写入到filelisttxt文件
function writeFile(data) {
    if(typeof data === 'object') data = JSON.stringify(data);
    data = 'var json = ' + data;
    fs.writeFile(rootPath + "/" + "filelist.js", data + '\n', function (err) {
        if (err) throw err;
        console.log("写入成功");
    });
}

//获取后缀名
function getdir(url) {
    var arr = url.split('.');
    var len = arr.length;
    return arr[len - 1];
}

File: handle_directory/handle.py
#coding:utf-8

# >>> [x for x in os.listdir('.') if os.path.isfile(x) and os.path.splitext(x)[1]=='.py']
# >>> [x for x in os.listdir('.') if os.path.isdir(x)]

import os
import json

pathList = []

def readDir(childList, pathName, dirName):
    pathDict = {
        'title': dirName,
        'urlList': [],
        'titleList': []
    }
    childList.append(pathDict)
    if os.path.isdir(pathName):
        # is 目录
        for x in os.listdir(pathName):
            readDir(pathDict['titleList'], os.path.join(pathName, x), x)
    if os.path.isfile(pathName):
        # is 文件
        urlDict = {
            'name': dirName,
            'url': pathName
        }
        pathDict['urlList'].append(urlDict)

def writeFile(data):
    with open('./pathlist.js', 'w', encoding='utf8') as f:
    	f.write(data)

# for x in os.listdir('.'):
#     readDir(pathList, x, x)

readDir(pathList, 'assets', 'assets')

# wObj = {
#     'json': pathList
# }
print(pathList)
pathList = pathList[0]['titleList']

pathList = json.dumps(pathList, ensure_ascii=False)

# print(pathList)

writeFile(str(pathList))
